src/blog/introducing-a-new-web-component-to-drag-resize-and-rotate.md

```diff
@@ -0,0 +1,76 @@
+---
+path: "/blog/introducing-a-new-web-component-to-drag-resize-and-rotate"
+date: "2020-03-03"
+title: "Introducing A New Web Component To Drag, Resize And Rotate"
+description: "Add drag, drop and resize capabilities to your web apps with this new Web Component compatible with or without any modern frameworks"
+tags: "#webdev #showdev #javascript #opensource"
+image: "https://cdn-images-1.medium.com/max/1600/1*1pqUKDcoxfwUQ6jN6tZ0xQ.png"
+canonical: https://medium.com/@david.dalbusco/syntax-highlighting-displayed-in-ubuntu-terminal-like-a7e9c310b504
+---
+
+![](https://cdn-images-1.medium.com/max/1600/1*1pqUKDcoxfwUQ6jN6tZ0xQ.png)
+
+We are happy to unleash today a new open source Web Component we have developed for our web editor for presentations, [DeckDeckGo](https://deckdeckgo.com), which can help you add a **drag**, **resize** and **rotate** feature to your applications or websites, regardless of whether or not you are using a modern web framework. Moreover it works out of the box without any extra JavaScript codes need.
+
+A demo is worth a thousand words, therefore here's what you can expect from it respectively how we are using it to let our users manipulate elements:
+
+![](https://cdn-images-1.medium.com/max/1600/1*_hmiwuTByWYJXuViGPBxFQ.gif)
+
+### Back Story
+
+Performances and dependencies  do really matters to us that’s why we tend to be a bit bundleophibic and let’s face it, we are also nerds 🤷. Coding is a much a job as a hobby. That’s why we do like to spend our evenings and weekends to develop these kind of stuffs.
+
+A couple of months ago, when we were brainstorming new ideas, [Nicolas](https://twitter.com/nasmattia) suggested to add a new template which would let users draw technical schema directly in our editor. This concept implied two complexities, preserving the aspect ratio of the content of the slide across devices and being able to manipulate (drag, resize and rotate) its content, that’s why it took us a bit to schedule and realize it.
+
+Moreover, even if at first glance it might looks easy to implement, we discovered while developing it, that the required maths were actually more challenging than expected and therefore obviously even more fun 😉
```

Comment:
```suggestion
While it may look easy to develop at first glance, during development we discovered that the required maths were actually more challenging than expected and therefore obviously even more fun 😉
```